src/Schema/Comparator.php

```diff
@@ -432,95 +434,76 @@ public function diffForeignKey(ForeignKeyConstraint $key1, ForeignKeyConstraint
     /**
      * Returns the difference between the columns
      *
-     * If there are differences this method returns $field2, otherwise the
-     * boolean false.
+     * If there are differences this method returns the changed properties as a
+     * string array, otherwise an empty array gets returned.
      *
      * @return string[]
      */
-    public function diffColumn(Column $column1, Column $column2)
+    public function diffColumn(Column $column1, Column $column2, AbstractPlatform $platform)
     {
         $properties1 = $column1->toArray();
         $properties2 = $column2->toArray();
 
-        $changedProperties = [];
+        // Do not compare the column names
+        unset($properties1['name'], $properties2['name']);
 
-        if (get_class($properties1['type']) !== get_class($properties2['type'])) {
-            $changedProperties[] = 'type';
-        }
+        $changedProperties = [];
 
-        foreach (['notnull', 'unsigned', 'autoincrement'] as $property) {
-            if ($properties1[$property] === $properties2[$property]) {
+        foreach (array_unique(array_merge(array_keys($properties1), array_keys($properties2))) as $key) {
```

Comment:
It might be simpler to do `array_keys(array_merge($properties1, $properties2))`. This way, there's no need for `array_unique()`.

Reply:
Done in 5dbb71531556225f189f5bc1cd26009b762d0c84